Add new action to run SonarQube analysis

gretzki opened the pull request:
Added new action to run SonarQube analysis with sonar-runner.

It's only a thin wrapper around sonar-runner binary, but I find it neat to have the checks and documentation integrated as separate fastlane action.

To be used inside Fastfile like this:

```
sonar(
  project_key: "name.gretzki.awesomeApp",
  project_version: "1.0",
  project_name: "iOS - AwesomeApp",
  sources_path: File.expand_path("../AwesomeApp")
)
```

Thanks,
Chris :smiley:

Commit:
Added `sonar` action to integrate sonar-runner for SonarQube analysis

Commit:
fixed travis and hound warnings

gretzki commented:
Well I don't see how the Travic-ci error is caused by my changes.
`bundle exec fastlane test` works fine when I run it locally with ruby 2.1.3

KrauseFx commented:
Sorry about the travis build - that's my fault, looking into the PR now :+1:

KrauseFx commented on lib/fastlane/actions/sonar.rb:
```diff
@@ -0,0 +1,86 @@
+      end
+
+      def self.verify_sonar_runner_binary
+        raise "You have to install sonar-runner using `brew install sonar-runner`".red unless `which sonar-runner`.to_s.length > 0 or !Helper.test?
```
Please use `||` instead of `or`, as the order is unclear otherwise :+1: Or even better: make it 3 lines instead of one

KrauseFx replied:
Why `Helper.test?`? You didn't add any tests for this action, right?

KrauseFx commented:
Thanks for the pull request @gretzki , that's a great addition. I added a comment before it's ready to be merged.

Also, could you add the example usage to `Actions.md`? :+1:

gretzki committed:
Adds sonar action to docs

Adds missing link to `other` section and inserted description of `sonar` action.

gretzki committed:
Removes superfluous Helper.test usage

gretzki commented:
Hi @KrauseFx,

thanks for the feedback. I've fixed the issue and added documentation as you suggested.
Hope it's all ok, now.

Btw: Congrats to this awesome tool!
Tried something quite similar (https://github.com/blckblt/blackbelt) in the old days, but made the mistake to never advertise it until it would be super shiny and bullet proof ;D
I'm just glad there's a tool that does the job and is supported by a quite large community :+1:

KrauseFx commented on docs/Actions.md:
```diff
@@ -23,6 +23,7 @@ import './path/to/other/Fastfile'
 - [Using git](#using-git)
 - [Using mercurial](#using-mercurial)
 - [Notifications](#notifications)
+- [Other](#other)
```
Why do we need `other` here?

KrauseFx commented:
Looks great, thanks for your great contribution @gretzki :rocket:
